### ISSUE TITLE
Homebridge-hue does not start if one light of a resourcelink is not reachable

### ISSUE THREAD

**Issue by ooii** [ebaauw/homebridge-hue#350]
Hi,

I'm running `deConz v2.5.25` (`v2.5.27` is not yet available on the ubuntu `/beta` repository) and `homebridge-hue v0.9.3`.
If one of the lights belonging to a `resourcelink` is not reachable (turned off from an electrical point of view) when `deconz` starts, then `homebridge-hue` won't start and complains as the attached log. As soon as the light is reachable again, everything works fine.
Note that while `homebridge-hue` is waiting for the gateway to initialise, this latter is working.

```
[2018-5-23 20:59:57] [Hue] conBee: /resourcelinks/6: /lights/42: not available
[2018-5-23 20:59:57] [Hue] conBee: gateway not yet initialised - wait 1 minute
```

**Comment by ebaauw** [comment]
This is by design.  The alternative would be that homebridge-hue starts without exposing the light, so it would be removed from HomeKit.

**Comment by ErnstEeldert** [comment]
While I understand the design choice from a homebridge-hue perspective, could you advice on a config strategy to minimize issues like this? For example, I have a whitelist setup as it's the easiest approach for me with a couple of multi-light 'lights', and some sensors bound to Phoscon created rules.

But, since the deconz rest api seems to be very aggressive in making devices no longer available, I run into this issue more than I would like. For example, a light was physically turned off by someone (the horror!), but all I know is the number, so I have to check all lights to see which one is the culprit. Not funny with over 40 lights.

But it gets worse, as I now have a 'missing' sensor. It's number 15, and I don't know which one it is.

So I guess what I am looking for is the deconz flag to keep a device visible until I hard delete it, with the appropriate availability flag set.

**Comment by ebaauw** [comment]
Historically, deCONZ handles sensors differently than lights.  Sensor resources are restored from the database on startup and marked reachable when the corrsponding node is discovered.  Lights are only restored after the node has been discovered.

> a light was physically turned off by someone (the horror!)

Send these barbarians back to the 20th century!

> but all I know is the number

That’s the onky info info available to homebridge-hue through the REST API (in the resourcelink).  Either renumber the lights, so the number makes sense (it helps when, like me,  you’re still recovering from OCD), or make a list (they’re managing companies in Excel, so why not a simple home; a simple `ph get -al /lights | grep /name:` will give you the info you need, once all lights have been found).  Without these, you’re best option is to look in the database.

>  I now have a 'missing' sensor

As far as I know, than can happen only when the sensor resource has ben deleted, through the REST API.  Again, best check the database - the record should still be there, with status “deleted”.

**Comment by cappadanna** [comment]
Every time I have tried to remove a light (or in this case an ikea outlet) I get the problem:

phoscon homebridge[20819]: [2019-12-11 20:30:04] [Hue] Phoscon-GW: /resourcelinks/9: /lights/4: not available
phoscon homebridge[20819]: [2019-12-11 20:30:04] [Hue] Phoscon-GW: gateway not yet initialised - wait 1 minute

I have tried waiting but nothing happens.
Homebridge won't work and I have to restore to my backup sd-card. What can I do if I want to remove an outlet but not mess up my homebridge?
